homeassistant/components/permobil/config_flow.py

```diff
@@ -0,0 +1,202 @@
+        errors: dict[str, str] = {}
+        if not self.p_api:
+            # create the api instance to use for validation of the user input
+            session = async_get_clientsession(self.hass)
+            self.p_api = MyPermobil(APPLICATION, session=session)
+
+        if user_input is not None:
+            # the user has entered their email in the first prompt
+            user_input[CONF_EMAIL] = user_input[CONF_EMAIL]
+            await validate_input(self.p_api, user_input)
+            self.data[CONF_EMAIL] = user_input[CONF_EMAIL]
+            _LOGGER.debug("Permobil: email %s", self.p_api.email)
+
+            await self.async_set_unique_id(self.data[CONF_EMAIL])
+            self._abort_if_unique_id_configured()
+
+        if errors or user_input is None:
+            # There was an error when the user entered their email
+            # or the user opened the flow for the first time
+            return self.async_show_form(
+                step_id="user", data_schema=GET_EMAIL_SCHEMA, errors=errors
+            )
+        # email prompt finished successfully, open the select region prompt
+        return await self.async_step_region()
+
+    async def async_step_region(self, user_input=None) -> FlowResult:
+        """Invoke when a user initiates a flow via the user interface."""
+        errors: dict[str, str] = {}
+        if not self.p_api:
+            # create the api for getting regions and sending the code
+            session = async_get_clientsession(self.hass)
+            self.p_api = MyPermobil(APPLICATION, session=session)
+
+        try:
+            if user_input is None:
+                # The user has opened the 2nd prompt for the first time
+                # fetch the list of regions names and urls from the api
+                # for the user to select from. [("name","url"),("name","url"),...]
+                self.region_names = await self.p_api.request_region_names()
+                _LOGGER.debug(
+                    "Permobil: region names %s",
+                    ",".join(list(self.region_names.keys())),
+                )
+
+            else:
+                # the user has selected their region name in the second prompt
+                # find the url for the selected region name
+                region_url = self.region_names[user_input[CONF_REGION]]  # KeyError
+                # set the region url in the api instance and in the entry
+                self.data[CONF_REGION] = region_url
+                self.p_api.set_region(region_url)
+                _LOGGER.debug("Permobil: region %s", self.p_api.region)
+                # tell backend to send code to the users email
+                # the code will be entered in the next prompt
+                await self.p_api.request_application_code()  # MyPermobilAPIException
+        except KeyError as err:
+            # the user has selected a region name that is not in the list (somehow)
+            errors["base"] = f"Pemobil: {err}"
```

Comment:
Please check https://developers.home-assistant.io/docs/data_entry_flow_index#validation and fix your `errors` dict

Reply:
The reson i do is this way is because i just want to forward whatever error message that the API returns to me. I agree its probably not the best idea to put it in the base, but on the other hand forwarding the error string from the API will make it easier to distinguish multiple errors that all originate from the same exception. Where should i forward the error message to instead?

Reply:
Print it in the log so the user can fetch it there.